### PR TITLE
Makes the game startup time significantly faster

### PR DESCRIPTION
**Pull request by Tastyfish**
Basically I removed a huge number of completely unnecessary sleeps/spawns, although for mining turfs (the main culprit) & headsets, refactored the setup so that they didn't need to spawn/sleep to sync up with things.
I also moved setting `sleep_offline` to the end of game setup, where it should have always been.

On my computer, DD went from pressing `Go` to `Finished object initializations` & the roundstart timer ticking in 55 seconds.

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets/secure/freezer.dm [hunk 1]
```diff
@@ -54,10 +52,8 @@
 
 	New()
 		..()
-		sleep(2)
-		for(var/i = 0, i < 4, i++)
+		for(var/i = 1 to 4)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/crates_lockers/closets/secure/freezer.dm]: IIRC, `for(var/i in X to Y)` is fastest

**Comment by Fox-McCloud** [comment]
Due to the nature of this being a performance only PR, and how much this impacts startup time, I see zero reasons for this to wait, and will be merging this immediately.

File: code/controllers/communications.dm [hunk 2]
```diff
@@ -205,9 +205,13 @@ var/const/RADIO_MAGNETS = "radio_magnet"
 var/const/RADIO_LOGIC = "radio_logic"
 
 var/global/datum/controller/radio/radio_controller
+var/global/list/headsets[0]
```
Comment (PJB3005) [inline code/controllers/communications.dm]: The `global` keyword does nothing on global variables.

File: code/controllers/communications.dm [hunk 3]
```diff
@@ -205,9 +205,13 @@ var/const/RADIO_MAGNETS = "radio_magnet"
...
 
 /hook/startup/proc/createRadioController()
 	radio_controller = new /datum/controller/radio()
+	for(var/A in headsets)
```
Comment (PJB3005) [inline code/controllers/communications.dm]: Doesn't `initialize()` get called on every object anyways so wouldn't this cause a second initialization of headsets?

Comment (Tastyfish) [inline code/controllers/communications.dm]: Only on objects directly in the world.

File: code/controllers/communications.dm (hunk 2 above)
Comment (Tastyfish) [inline code/controllers/communications.dm]: I know, just matching the weird convention already established for the other global lists.

File: code/controllers/communications.dm (hunk 3 above)
Comment (PJB3005) [inline code/controllers/communications.dm]: Yes but in that case wouldn't all headsets be created *after* the radio controller is made so it doesn't matter?

`createRadioControlle()` is called by https://github.com/ParadiseSS13/Paradise/blob/master/code/world.dm#L49

`initialize()` gets called later down the line through `master_controller.setup()`

Any atom is always `in world`, so the initializations would catch them no matter what.

Comment (Tastyfish) [inline code/controllers/communications.dm]: In retrospect I'm not sure what I was making extra consideration for by running initialize() here.

**Pull request merged**